src/backend/aspen/app/views/sample.py

```diff
@@ -230,3 +263,61 @@ def samples():
             results.append(returned_sample_data)
 
         return jsonify({SAMPLE_KEY: results})
+
+
+@application.route("/api/samples/create", methods=["POST"])
+@requires_auth
+def create_sample():
+    with session_scope(application.DATABASE_INTERFACE) as db_session:
+        profile: Mapping[str, str] = session["profile"]
+
+        user: User = (
+            get_usergroup_query(db_session, profile["user_id"])
+            .options(joinedload(User.group))
+            .one()
+        )
+        request_data = request.get_json()
+
+        for data in request_data:
+            data_ok: bool
+            missing_fields: Optional[list[str]]
+            unexpected_fields: Optional[list[str]]
```

Comment:
I don't think you need to instantiate these values before the check_data call

Reply:
apparently this is how you add type hints for when you have a tuple return value, we can remove the type hints though, i personally find them a little too much sometimes lol

Reply:
oh interesting, let's leave them then! extra info : D